src/lib/utils/parsing.cpp

```diff
@@ -27,15 +27,16 @@ uint16_t to_uint16(const std::string& str)
    return static_cast<uint16_t>(x);
    }
 
-uint32_t to_u32bit(const std::string& str)
+uint32_t to_u32bit(std::string_view str_view)
    {
+   const std::string str(str_view);
```

Comment:
One could consider using [`std::from_chars()`](https://en.cppreference.com/w/cpp/utility/from_chars) to avoid the copy. Though, the API seems a bit low-level for my taste. Still worth considering.

Reply:
I just realized this whole function is actually exceptionally inefficient (not that it really matters) since we scan the string twice. I'll open another PR here.